### PR TITLE
official-images cleanup and improvements

### PR DESCRIPTION
**Pull request by yosifkit**
- moved to be `FROM buildpack-deps:jessie-scm`,  this gives users access to tools like `git` and `curl`, without adding in all the extra c-libs that regular `buildpack-deps` has.
  - are C library dependencies in perl6 modules very common?  if so, we should switch this to just use `buildpack-deps` and consider adding a `:slim` version.
- install deps that are needed for build and remove them in the same layer to slim the image
- remove the source (if the source dir is useful in the image I can pull it out of the `rm`)
- swap some ordering for better docker caching of layers

### CODE REVIEW

**Comment by hoelzro** [comment]
@yosifkit thanks for the pr! regarding c library dependencies, are you asking if perl 6 modules need to compile c libraries, or if they reference libraries under /usr/lib?

**Comment by tianon** [comment]
I think a good example would be `XML::LibXML` for Perl 5, which is a wrapper around `libxml2`.  Is this as common in Perl 6 as it is in 5, or is it more like the Go world, where the majority of libraries get re-implemented natively?

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
-WORKDIR /root/rakudo-star-${rakudo_version}
+ENV rakudo_version=2015.03
 
-RUN perl Configure.pl --prefix=/usr --gen-moar
-RUN make install
+RUN buildDeps=' \
+        gcc \
+        libc6-dev \
+        libencode-perl \
+        make \
```
Comment (yosifkit) [inline Dockerfile:12]: More specifically, are there any of these build dependencies that should be kept for users of the image?

Would users benefit from having all the libraries installed that are part of the [`buildpack-deps`](https://github.com/docker-library/buildpack-deps/blob/master/jessie/Dockerfile)?

Comment (hoelzro) [inline Dockerfile:12]: I think that enough of those are required by the modules in Rakudo Star that they should be left in.  I'll try out all of the modules to see if other relevant libraries are missing.

**Comment by hoelzro** [comment]
Thanks for the cleanup @yosifkit, and for the input @tianon!

**Comment by hoelzro** [comment]
Ok, I just tried all of the Star modules, and they all look good on the generated image!

**Pull request merged**